client/lib/url-search/index.js

```diff
@@ -2,37 +2,94 @@
  * External dependencies
  */
 import React from 'react';
+import debugModule from 'debug';
+import page from 'page';
+import url from 'url';
+import { pick } from 'lodash';
 
 /**
  * Internal dependencies
  */
-import { doSearch, getSearchOpen } from 'lib/mixins/url-search';
+const debug = debugModule( 'calypso:url-search' );
 
-export default function( Component ) {
-	const componentName = Component.displayName || Component.name || '';
+/**
+ * Function for constructing the url to page to. e.g.
+ * { uri: 'google.com', search:'hello' } --> 'google.com?s=hello'
```

Comment:
What do you think is the best thing to put there? Give more examples of potential usage?